datafusion/functions/src/regex/regexpreplace.rs

```diff
@@ -382,49 +382,32 @@ where
     }
 }
 
-#[expect(clippy::needless_pass_by_value)]
```

Comment:
I wanted to fix this lint, so realised could do a wider refactor to eliminate this function and simplify the code